src/Admin/Filesystem/FileAdmin.php

```diff
@@ -0,0 +1,294 @@
+    {
+        $isCreation = null === $this->getSubject() || null === $this->getSubject()->getId();
+
+        $formMapper
+            ->add('name', TextType::class, [
+                'label' => 'Titre',
+                'filter_emojis' => true,
+            ])
+            ->add('file', FileType::class, [
+                'label' => 'Ajouter un fichier',
+                'required' => false,
+                'help' => 'Le fichier ne doit pas dépasser 5 Mo.'.(!$isCreation ? '' : ' Laissez vide pour ne pas modifier'),
+            ])
+            ->add('externalLink', UrlType::class, [
+                'label' => 'URL',
+                'required' => false,
+            ])
+            ->add('parent', FileParentType::class, [
+                'label' => 'Dossier',
+                'help' => 'Saisisez le nom pour voir les dossiers existants. Si vous ne trouvez pas le dossier souhaiter mettez le nom complet d\'un nouveau dossier.',
+            ])
+            ->add('permissions', CollectionType::class, [
+                'entry_type' => FilePermissionType::class,
+                'required' => false,
+                'label' => 'Permissions',
+                'allow_add' => true,
+                'allow_delete' => true,
+                'by_reference' => false,
+            ])
+            ->add('displayed', null, [
+                'label' => 'Affiché',
+            ])
+        ;
+
+        $formMapper->getFormBuilder()
+            ->addEventListener(FormEvents::SUBMIT, [$this, 'submit'])
+            ->addEventSubscriber(new FileParentListener($this->repository))
+        ;
+    }
+
+    protected function configureListFields(ListMapper $listMapper)
+    {
+        $listMapper
+            ->add('fullPath', null, [
+                'label' => 'Dossier/Document',
+            ])
+            ->add('permissions', null, [
+                'label' => 'Rôles',
+                'template' => 'admin/filesystem/list_permissions.html.twig',
+            ])
+            ->add('createdAt', null, [
+                'label' => 'Date d\'ajout',
+            ])
+            ->add('createdBy', null, [
+                'label' => 'Créé par',
+            ])
+            ->add('updatedAt', null, [
+                'label' => 'Date de modification',
+            ])
+            ->add('updatedBy', null, [
+                'label' => 'Modifié par',
+            ])
+            ->add('displayed', null, [
+                'label' => 'Est affiché ?',
+                'editable' => true,
+            ])
+            ->add('_action', null, [
+                'virtual_field' => true,
+                'template' => 'admin/filesystem/list_actions.html.twig',
+            ])
+        ;
+    }
+
+    public function submit(FormEvent $event): void
+    {
+        /** @var File $file */
+        $file = $event->getData();
+        $uploadedFile = $file->getFile();
+
+        $this->fileManager->update($file, $uploadedFile);
```

Comment:
if an error of form validation appears we will keep the file on storage? 
what about saving the updated file on `postPersist` instead?